src/index.js

```diff
@@ -665,74 +663,74 @@ const initialize = async () => {
      * ERC721 Token
      */
 
-    deployCollectiblesButton.onclick = async () => {
-      collectiblesStatus.innerHTML = 'Deploying';
+    deployNFTsButton.onclick = async () => {
+      nftsStatus.innerHTML = 'Deploying';
 
       try {
-        collectiblesContract = await collectiblesFactory.deploy();
-        await collectiblesContract.deployTransaction.wait();
+        nftsContract = await nftsFactory.deploy();
+        await nftsContract.deployTransaction.wait();
       } catch (error) {
-        collectiblesStatus.innerHTML = 'Deployment Failed';
+        nftsStatus.innerHTML = 'Deployment Failed';
         throw error;
       }
 
-      if (collectiblesContract.address === undefined) {
+      if (nftsContract.address === undefined) {
         return;
       }
 
       console.log(
-        `Contract mined! address: ${collectiblesContract.address} transactionHash: ${collectiblesContract.deployTransaction.hash}`,
+        `Contract mined! address: ${nftsContract.address} transactionHash: ${nftsContract.deployTransaction.hash}`,
       );
-      collectiblesStatus.innerHTML = 'Deployed';
+      nftsStatus.innerHTML = 'Deployed';
       mintButton.disabled = false;
       mintAmountInput.disabled = false;
     };
 
+    watchNFTsButton.onclick = async () => {
+      const currentTokenId = await nftsContract.currentTokenId();
+      const nftsContractAddress = nftsContract.address;
+      let watchNftsResult;
+      try {
+        watchNftsResult = await ethereum.sendAsync(
+          Array.from({ length: currentTokenId }, (_, i) => i + 1).map(
+            (tokenId) => {
+              return {
+                method: 'wallet_watchAsset',
+                params: {
+                  type: 'ERC721',
+                  options: {
+                    address: nftsContractAddress,
+                    tokenId,
+                  },
+                },
+              };
+            },
+          ),
+        );
+      } catch (error) {
+        console.error(error);
+      }
+      console.log(watchNftsResult);
+    };
```

Comment:
This change is slightly unrelated. It is simply moving this onclick handler outside of the `mintButton` onclick handler so that it can be used to watch already minted NFTs when you pass a contract address to the dapp via url query string (without calling a new mint in that session).